src/mlpack/core/dists/gamma_distribution.cpp

```diff
@@ -0,0 +1,76 @@
+/**
+ * @file gamma_distribution.cpp
+ * @author Yannis Mentekidis
+ *
+ * Implementation of the methods of GammaDistribution.
+ */
+#include "gamma_distribution.hpp"
+
+#include <boost/math/special_functions/digamma.hpp>
+#include <boost/math/special_functions/trigamma.hpp>
```

Comment:
Yep, my mistake, it was added on Boost 1.58. I misunderstood the article I linked before, when scanning it I thought there was a relationship so you could compute trigamma(x) = f(digamma(x)) but instead it is trigamma(x) = f(trigamma(x-1)).

I think the best option is to backport the source code from boost... I'll start examining how to do that, hopefully it will be as straightforward as copying a file...